src/handlers/payout/action.ts

```diff
@@ -127,78 +168,236 @@ export const handleIssueClosed = async () => {
-    return `Permit generation skipped since issue's bounty is higher than ${paymentPermitMaxPrice}`;
+    return { error: `Permit generation skipped since issue's bounty is higher than ${incentivesCalculation.paymentPermitMaxPrice}` };
   }
 
   // if bounty hunter has any penalty then deduct it from the bounty
-  const penaltyAmount = await getPenalty(assignee.login, payload.repository.full_name, paymentToken, networkId.toString());
+  const penaltyAmount = await getPenalty(
+    assigneeLogin,
+    incentivesCalculation.payload!.repository.full_name,
+    incentivesCalculation.paymentToken!,
+    incentivesCalculation.networkId!.toString()
+  );
   if (penaltyAmount.gt(0)) {
     logger.info(`Deducting penalty from bounty`);
     const bountyAmount = ethers.utils.parseUnits(priceInEth.toString(), 18);
     const bountyAmountAfterPenalty = bountyAmount.sub(penaltyAmount);
     if (bountyAmountAfterPenalty.lte(0)) {
-      await removePenalty(assignee.login, payload.repository.full_name, paymentToken, networkId.toString(), bountyAmount);
+      await removePenalty(
+        assigneeLogin,
+        incentivesCalculation.payload!.repository.full_name,
+        incentivesCalculation.paymentToken!,
+        incentivesCalculation.networkId!.toString(),
+        bountyAmount
+      );
       const msg = `Permit generation skipped because bounty amount after penalty is 0`;
       logger.info(msg);
-      return msg;
+      return { error: msg };
     }
     priceInEth = new Decimal(ethers.utils.formatUnits(bountyAmountAfterPenalty, 18));
   }
 
-  const { txData, payoutUrl } = await generatePermit2Signature(recipient, priceInEth, issue.node_id, assignee.node_id, "ISSUE_ASSIGNEE");
-  const tokenSymbol = await getTokenSymbol(paymentToken, rpc);
-  const shortenRecipient = shortenEthAddress(recipient, `[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]`.length);
-  logger.info(`Posting a payout url to the issue, url: ${payoutUrl}`);
-  const comment =
-    `#### Task Assignee Reward\n### [ **[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n` + "```" + shortenRecipient + "```";
-  const permitComments = comments.filter((content) => content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot);
-  if (permitComments.length > 0) {
-    logger.info(`Skip to generate a permit url because it has been already posted`);
-    return `Permit generation skipped because it was already posted to this issue.`;
-  }
-  await deleteLabel(issueDetailed.priceLabel);
-  await addLabelToIssue("Permitted");
-  await savePermitToDB(assignee.id, txData);
-  if (penaltyAmount.gt(0)) {
-    await removePenalty(assignee.login, payload.repository.full_name, paymentToken, networkId.toString(), penaltyAmount);
+  const account = await getWalletAddress(assigneeLogin);
+
+  return {
+    error: "",
+    user_id: incentivesCalculation.assignee!.node_id,
```

Comment:
```suggestion
    user_id: incentivesCalculation.assignee!.id,
```
we use id, not node_id

Reply:
> ```suggestion
>     user_id: incentivesCalculation.assignee!.id,
> ```
> we use id, not node_id

Oh, okay.. I'd use that instead

Reply:
> we use id, not node_id

Its always been `node_id`, you can check the current development branch [here](https://github.com/ubiquity/ubiquibot/blob/08a2dfb1ffbf426fd984a97fe583a310a7c1e43a/src/handlers/payout/action.ts#L186)

Reply:
Yes, it doesn't much difference, but I think we should at least try to keep it the same everywhere.

Reply:
In the early days of the github REST API all entities (issue, user, comment, etc...) used the `id` property. As far as I understand `issue.id`, `user.id` and `comment.id` can intersect while the `node_id` property is globally unique across all entities. That is why [here](https://github.com/ubiquity/ubiquibot/blob/08a2dfb1ffbf426fd984a97fe583a310a7c1e43a/src/handlers/payout/action.ts#L186) `issue.node_id` was used. 

The 2nd useful thing regarding the `node_id` property is that some APIs are missing from the github REST API while exist in the graphql API where you can fetch entities by `node_id`.

That is why I think it is better to use `node_id` whenever possible compared to the `id` property.